internal/tss/channels/rpc_caller_channel.go

```diff
@@ -0,0 +1,81 @@
+package channels
+
+import (
+	"context"
+
+	"github.com/alitto/pond"
+
+	"github.com/stellar/go/support/log"
+	"github.com/stellar/wallet-backend/internal/entities"
+	"github.com/stellar/wallet-backend/internal/tss"
+	"github.com/stellar/wallet-backend/internal/tss/router"
+	"github.com/stellar/wallet-backend/internal/tss/services"
+	"github.com/stellar/wallet-backend/internal/tss/store"
+)
+
+type RPCCallerChannelConfigs struct {
+	TxManager     services.TransactionManager
+	Router        router.Router
+	Store         store.Store
+	MaxBufferSize int
+	MaxWorkers    int
+}
+
+type rpcCallerPool struct {
+	Pool      *pond.WorkerPool
+	TxManager services.TransactionManager
+	Router    router.Router
+	Store     store.Store
+}
+
+var RPCCallerChannelName = "RPCCallerChannel"
+
+func NewRPCCallerChannel(cfg RPCCallerChannelConfigs) *rpcCallerPool {
+	pool := pond.New(cfg.MaxBufferSize, cfg.MaxWorkers, pond.Strategy(pond.Balanced()))
+	return &rpcCallerPool{
+		Pool:      pool,
+		TxManager: cfg.TxManager,
+		Store:     cfg.Store,
+		Router:    cfg.Router,
+	}
+
+}
+
+func (p *rpcCallerPool) Send(payload tss.Payload) {
+	p.Pool.Submit(func() {
+		p.Receive(payload)
+	})
+}
+
+func (p *rpcCallerPool) Receive(payload tss.Payload) {
+
+	ctx := context.Background()
+	// Create a new transaction record in the transactions table.
+	err := p.Store.UpsertTransaction(ctx, payload.WebhookURL, payload.TransactionHash, payload.TransactionXDR, tss.RPCTXStatus{OtherStatus: tss.NewStatus})
+
+	if err != nil {
+		log.Errorf("%s: Unable to upsert transaction into transactions table: %e", RPCCallerChannelName, err)
+		return
+	}
+	rpcSendResp, err := p.TxManager.BuildAndSubmitTransaction(ctx, RPCCallerChannelName, payload)
+
+	if err != nil {
+		log.Errorf("%s: Unable to sign and submit transaction: %e", RPCCallerChannelName, err)
```

Comment:
```suggestion
		log.Errorf("%s: unable to sign and submit transaction: %w", RPCCallerChannelName, err)
```

Reply:
Done

Reply:
You forgot to fix the `%e`, this is what it's printing now: https://go.dev/play/p/wp-a11Jbr3C

Reply:
Same for the other two errors below